docsource/Makefile

```diff
@@ -0,0 +1,20 @@
+# Minimal makefile for Sphinx documentation
+#
+
+# You can set these variables from the command line.
+SPHINXOPTS    =
+SPHINXBUILD   = sphinx-build
+SPHINXPROJ    = KBaseSearchEngine
+SOURCEDIR     = .
+BUILDDIR      = build
+
+# Put it first so that "make" without argument is like "make help".
+help:
+	@$(SPHINXBUILD) -M help "$(SOURCEDIR)" "$(BUILDDIR)" $(SPHINXOPTS) $(O)
+
+.PHONY: help Makefile
+
+# Catch-all target: route all unknown targets to Sphinx using the new
+# "make mode" option.  $(O) is meant as a shortcut for $(SPHINXOPTS).
+%: Makefile
+	@$(SPHINXBUILD) -M $@ "$(SOURCEDIR)" "$(BUILDDIR)" $(SPHINXOPTS) $(O)
```

Comment:
OK, this file reminds me of why I hate make+bash so much. There's eight zillion magic symbols whose semantics have nothing to do with their representation, I don't work in make+bash land enough to remember them, and they're impossible to google. Even a very simple file like this is unreadable after 20m of googling. I'm pretty sure I'm not the only person that feels this way...

So: can you please just go completely overboard in documenting this and what the thingies mean? In particular:
@$ vs $@
why SOURCEDIR and BUILDDIR are in quotes and SPHINXOPTS is not
what the expected values for the input to the make file are (i.e. what is expected to wind up in $@)
why you have the catchall target rather than a plain old make build or something (the value of % is ignored, right?)
What's the deal with $(O). I see the doc line, I just don't get it.

Reply:
This makefile was generated using 'sphinx-quickstart' and we should treat is as opaque from the perspective of our project. I have added a readme giving the necessary background behind the documentation process.